src/commands/cmd_server.cc

```diff
@@ -1123,6 +1126,65 @@ class CommandRdb : public Commander {
   uint32_t db_index_ = 0;
 };
 
+class CommandAnalyze : public Commander {
+ public:
+  Status Parse(const std::vector<std::string> &args) override {
+    if (args.size() <= 1) return {Status::RedisExecErr, errInvalidSyntax};
+    for (int i = 1; i < args.size(); ++i) {
+      command_args_.push_back(args[i]);
+    }
+    return Status::OK();
+  }
+  Status Execute(Server *srv, Connection *conn, std::string *output) override {
+    auto commands = redis::CommandTable::Get();
+    auto cmd_iter = commands->find(util::ToLower(command_args_[0]));
+    if (cmd_iter == commands->end()) {
+      // unsupported redis command
+      return {Status::RedisExecErr, errInvalidSyntax};
+    }
+    auto redis_cmd = cmd_iter->second;
+    auto cmd = redis_cmd->factory();
+    cmd->SetAttributes(redis_cmd);
+    cmd->SetArgs(command_args_);
+
+    auto s = cmd->Parse(command_args_);
```

Comment:
Should we check the arity before `Parse`? Since wrong `command_args_` might make our program crash

Reply:
We need to reuse the code in ExecuteCommand rather than write them again here.

Reply:
I just pushed a newer version which added the arity checking logic. As for reusing / refactoring the code in `ExecuteCommand`, maybe it's better to do it in a separate pr? As the `ExecuteCommand` code needs to be dissected into several chunks for reusing purpose.